File: smart_home_app/smart_home_core/lib/src/models/device.dart
import 'dart:io';

import 'device_type.dart';

abstract class Device {
  DeviceType type;
  String hostname;
  String uuid;

  String name;

  InternetAddress? ipAddress;

  Device({required this.type, required this.hostname, required this.uuid, required this.name, this.ipAddress});
  Device.virtual({required this.type, required this.name}) : hostname = '#virtual', uuid = '';

  bool get isVirtual => hostname == "#virtual";
}

File: smart_home_angular/src/app/core/models/device.ts
import { DeviceType } from './device_type';

export class Device {
  public id!: string;
  public name!: string;
  public type!: DeviceType;
  public hostname!: string;

  constructor(id: string, name: string, type: DeviceType, hostname: string) {
    this.id = id;
    this.name = name;
    this.type = type;
    this.hostname = hostname;
  }
}
